lmos-runtime-core/src/main/kotlin/ai/ancf/lmos/runtime/core/inbound/ConversationHandler.kt

```diff
@@ -4,46 +4,49 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-package ai.ancf.lmos.runtime.core.service.inbound
+package ai.ancf.lmos.runtime.core.inbound
 
-import ai.ancf.lmos.runtime.core.constants.ApiConstants.Headers.SUBSET
+import ai.ancf.lmos.runtime.core.LmosRuntimeConfig
+import ai.ancf.lmos.runtime.core.cache.LmosRuntimeTenantAwareCache
+import ai.ancf.lmos.runtime.core.constants.LmosRuntimeConstants.SUBSET
 import ai.ancf.lmos.runtime.core.model.Agent
 import ai.ancf.lmos.runtime.core.model.AssistantMessage
 import ai.ancf.lmos.runtime.core.model.Conversation
-import ai.ancf.lmos.runtime.core.service.cache.LmosRuntimeTenantAwareCache
 import ai.ancf.lmos.runtime.core.service.outbound.AgentClientService
 import ai.ancf.lmos.runtime.core.service.outbound.AgentRegistryService
 import ai.ancf.lmos.runtime.core.service.outbound.AgentRoutingService
 import org.slf4j.LoggerFactory
 
-public interface ConversationService {
-    suspend fun processConversation(
+interface ConversationHandler {
+    suspend fun handleConversation(
         conversation: Conversation,
         conversationId: String,
         tenantId: String,
         turnId: String,
     ): AssistantMessage
 }
 
-class DefaultConversationService(
+class DefaultConversationHandler(
     private val agentRegistryService: AgentRegistryService,
     private val agentRoutingService: AgentRoutingService,
     private val agentClientService: AgentClientService,
     private val lmosRuntimeTenantAwareCache: LmosRuntimeTenantAwareCache<String>,
-) : ConversationService {
-    private val log = LoggerFactory.getLogger(DefaultConversationService::class.java)
+    private val lmosRuntimeConfig: LmosRuntimeConfig,
+) : ConversationHandler {
+    private val log = LoggerFactory.getLogger(DefaultConversationHandler::class.java)
 
-    override suspend fun processConversation(
+    override suspend fun handleConversation(
         conversation: Conversation,
         conversationId: String,
         tenantId: String,
         turnId: String,
     ): AssistantMessage {
-        val subset = lmosRuntimeTenantAwareCache.get(tenantId, SUBSET, conversationId)
+        var subset = lmosRuntimeTenantAwareCache.get(tenantId, SUBSET, conversationId)
```

Comment:
I still don't get why subset is cached. 
Why can't you just cache the `routingInformation` and use the tenantId + conversationId as a cache key?
The RoutingInformation must not be cached for the whole tenant, channel.

Reply:
changes completed